agents/mcts.py

```diff
@@ -129,24 +133,26 @@ def __init__(self, expand_policy: ExpandPolicy = default_expand_policy,
 
     def __check_terminal(self, board: Board, is_player_opponent: bool) -> float:
         board = board.reshape(self._board_size, self._board_size)
-        status = self._eval.evaluate(board, -1 if is_player_opponent else 1)
-        if np.isinf(status):
-            if (status < 0) == is_player_opponent:
-                return 1
+        sign = -1 if is_player_opponent else 1
+        evaluation = self._eval.evaluate(board, -1 if is_player_opponent else 1)
+        if np.isinf(evaluation):
+            if evaluation > 0:
+                evaluation = MAX_EVALUATION
             else:
-                return -1
+                evaluation = -MAX_EVALUATION
+            return sign * evaluation
         return None
 
     def __eval(self, board: Board, is_player_opponent: bool) -> float:
         board = board.reshape(self._board_size, self._board_size)
-        mult = -1 if is_player_opponent else 1
-        status = self._eval.evaluate(board, -1 if is_player_opponent else 1) * mult
-        status -= self._base_status
-        if status > self._epsilon:
-            return 1
-        elif status < -self._epsilon:
-            return -1
-        return 0
+        sign = -1 if is_player_opponent else 1
+        evaluation = self._eval.evaluate(board, -1 if is_player_opponent else 1) * sign
+        if np.isinf(evaluation):
+            if evaluation > 0:
+                evaluation = MAX_EVALUATION
+            else:
+                evaluation = -MAX_EVALUATION
+        return sign * evaluation
```

Comment:
Should you multiple here evaluation twice by a sign?

Reply:
Shouldn't, change this with some other changes